cmd/consensus/main.go

```diff
@@ -173,7 +176,21 @@ func main() {
 		}).
 		Module("random beacon key", func(node *cmd.FlowNodeBuilder) error {
 			privateDKGData, err = loadDKGPrivateData(node.BaseConfig.BootstrapDir, node.NodeID)
-			return err
+			if err != nil {
+				return err
+			}
```

Comment:
- Are these checks to be performed right after attempting to load the key for the root block?
- In fact, why do we need to load the key for the epoch corresponding to the root block if state.Final() is already further ahead (potentially in a different epoch)?
- In your second bullet point, shouldn't we be looking for the key corresponding to  `state.Final().Epochs().Next().Counter()` instead of `state.Final().Epochs().Current().Counter()`?

Reply:
> Are these checks to be performed right after attempting to load the key for the root block?

Yes, they would be after we load the key for the root block's epoch. The important thing is that we detect being in a bad state as soon as possible and crash, ideally before completing the startup process. 

We could do this here as part of loading the root block's DKG key. Another place it could make sense would be when instantiating the component that manages cycling DKG keys at epoch boundaries (https://github.com/dapperlabs/flow-go/issues/5275). This component must be aware of the protocol state in order to decide which epoch a block is in, so we would have all the information we need there to perform these sanity checks.

> In fact, why do we need to load the key for the epoch corresponding to the root block if state.Final() is already further ahead (potentially in a different epoch)?

We don't, we could add that check to avoid reading the file. I think the logic is a bit simpler without adding this check, but I don't have a strong opinion on this. 

> In your second bullet point, shouldn't we be looking for the key corresponding to state.Final().Epochs().Next().Counter() instead of state.Final().Epochs().Current().Counter()?

Yes sorry you are right. If we are a valid consensus node in the next epoch, and have reached the `EpochCommitted` phase in the current epoch, and have not inserted our DKG private keyshare for the next epoch, we are in a bad state. I updated my comment as well.

Reply:
Thanks,
I put the checks here for now. Will move them to another module in the next PR for [#5275](https://github.com/dapperlabs/flow-go/issues/5275)